src/parser.js

```diff
@@ -1218,10 +1234,20 @@ export class GenericParser extends Tokenizer {
           init: null,
         }));
 
-      case 'ObjectExpression':
+      case 'ObjectExpression': {
+        let last = node.properties.length > 0 ? node.properties[node.properties.length - 1] : void 0;
+        if (last != null && last.type === 'SpreadProperty') {
+          return this.copyNode(node, new AST.ObjectAssignmentTarget({
+            properties: node.properties.slice(0, -1).map(e => e && this.transformDestructuringWithDefault(e)),
+            rest: this.transformDestructuringWithDefault(last.expression),
```

Comment:
Why is this `transformDestructuringWithDefault`? It looks like that would allow `({...a = 0} = 1)`, which it should not. (We should also have a test asserting that that's not allowed.)

Reply:
We need this to parse `({a = 0})`, but added a check in `parseObjectExpression` to throw an error if `}` followed by `=`

Reply:
`({a = 0})` should not parse.